Examples/tvOSMultiplePlayersTemplate/tvOSMultiplePlayersTemplate/PlayersCollectionViewController.swift

```diff
@@ -0,0 +1,140 @@
+
+//  Copyright © 2017 One by Aol : Publishers. All rights reserved.
+
+import UIKit
+import OneMobileSDK
+import AVFoundation
+import AVKit
+
+class PlayersCollectionViewController: UICollectionViewController {
+    
+    var dataSource: PlayersCollectionDTO?
+    var provider = OneSDK.Provider.default
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        guard let dataSource = self.dataSource else { return }
+        
+        for playerData in dataSource.playersData {
+            self.provider.context.extra = ["noAds": true]
+            self.provider.getSDK()
+                .then { $0.videoProvider.getVideosBy(videoIDs: [playerData.videoId]) }
+                .dispatch(on: .main)
+                .onComplete(callback: handle)
+        }
+    }
+    
+    func handle(result: Result<VideoProvider.Response>) {
+        
+        switch result {
+        case .value(let response):
+            
+            self.provider.getSDK()
```

Comment:
For some reasons you are requesting players twice, once in `viewDidLoad` and once here.